source/build.py

```diff
@@ -0,0 +1,168 @@
+                     level=logging.DEBUG,
+                     format='%(levelname)s: %(message)s [%(asctime)s]',
+                     datefmt='%m/%d/%Y %I:%M:%S %p' )
+
+cwd = str(os.getcwd())
+lsResults = os.listdir(cwd)
+
+isMkdocsInstalled = (
+    subprocess.check_output(['mkdocs', '--version'])[0:15] == 'mkdocs, version')
+
+googleTagManagerContainerId = 'GTM-MNRH4J'
+
+googleTagManagerScriptLines = """
+<!-- Google Tag Manager -->
+  <noscript>
+    <iframe src="//www.googletagmanager.com/ns.html?id={containerId}"
+    height="0" width="0" style="display:none;visibility:hidden"></iframe>
+  </noscript>
+  <script>
+    (function(w,d,s,l,i){{w[l]=w[l]||[];w[l].push({{'gtm.start':
+    new Date().getTime(),event:'gtm.js'}});var f=d.getElementsByTagName(s)[0],
+    j=d.createElement(s),dl=l!='dataLayer'?'&l='+l:'';j.async=true;j.src=
+    '//www.googletagmanager.com/gtm.js?id='+i+dl;f.parentNode.insertBefore(j,f);
+    }})(window,document,'script','dataLayer','{containerId}');
+  </script>
+<!-- End Google Tag Manager -->
+ """.format(containerId = googleTagManagerContainerId)
+# """.format(containerId = googleTagManagerContainerId).splitlines()
+
+
+# ============================================================================
+
+
+class PostMkdocsParser(HTMLParser):
+
+    def __init__(self, htmlFileFullPath):
+        self.reset()
+        self.markers          = set([])
+        self.targetIsFullHTML = False
+        self.targetFullPath   = htmlFileFullPath
+        self.pruneLangs    = set([])
+        if '/en/' in self.targetFullPath:
+            self.pruneLangs.add('ja')
+        if '/ja/' in self.targetFullPath:
+            self.pruneLangs.add('en')
+
+    def handle_decl(self, data):
+        if data == "DOCTYPE html":
+            self.targetIsFullHTML = True
+
+            if not ('/en/' in self.targetFullPath or '/ja/' in self.targetFullPath):
+                self.pruneLangs.add('en')
+                self.pruneLangs.add('ja')
+
+    def handle_starttag(self, tag, attrs):
+        position = self.getpos()[0]
+        if self.targetIsFullHTML and tag == 'body':
+            # mark for injection of Google Tag Manager Script.
+            self.markers.add(('injectTagManager', position))
+
+        if ( tag == 'a' and (('href', '..') in attrs) and
+                            (('class', '') in attrs)
+           ):
+                # mark for removal: unelegant blank TOC item.
+                self.markers.add(('prune', position))
+
+        for attr in attrs:
+            if isPruneLangElement(self, tag, attr):
+                # mark for removal: TOC items of the other language.
+                self.markers.add(('prune', position))
+
+                # retroactively prevent removal of main links from
+                #  top level page to each language's index page.
+                if isExceptionTopIndex(self, attr):
+                    self.markers.remove(('prune', position))
+
+def isPruneLangElement(self, tag, attr):
```

Comment:
if you place this function out of the class, you shouldn't pass `self`, because function don't know the property of self. instead, give `pruneLangs` as arguement.